post-install-setup.sh

```diff
@@ -0,0 +1,237 @@
+#
+# No User Serviceable Parts Beyond This Point
+#	Qualified service personal only
+#
+# ----------------------------------------------------------------------------
+
+#
+# Check if the users in question all actually exist
+#
+
+which getent &> /dev/null
+
+if [[ "$?" != "0" ]]
+then
+	echo "command \`which\` is not installed or available in the path. Can't perform checks. This is a fatal error."
+	exit
+fi
+
+getent passwd "${_LIOTA_ROOTUSER}" &> /dev/null
+
+if [[ "$?" != "0" ]]
+then
+	echo "'Root' User ${_LIOTA_ROOTUSER} is not setup, this is odd and needs to be resolved. This is a fatal error."
+	exit
+fi
+
+getent passwd "${_LIOTA_USER}" &> /dev/null
+
+if [[ "$?" != "0" ]]
+then
+	echo "Liota User ${_LIOTA_USER} is not setup, this is odd and needs to be resolved. This is a fatal error."
+	exit
+fi
+
+getent group "${_LIOTA_USER}" &> /dev/null
+
+if [[ "$?" != "0" ]]
+then
+	echo "Liota Group ${_LIOTA_USER} is not setup, this is odd and needs to be resolved. This is a fatal error."
+	exit
+fi
+
+
+#
+# Configuration directory
+#
+
+if [[ -e "${_LIOTA_ETCDIR}" ]]
+then
+	echo "ERROR: ${_LIOTA_ETCDIR} already exists, and assuming things are setup already or not but automatic running will not proceed. This is a fatal error."
+	exit
+fi
+
+# Deal with creating the directory, if it doesn't exist
+if [[ ! -d "${_LIOTA_ETCDIR}" ]]
+then
+	if [[ -e "${_LIOTA_ETCDIR}" ]]
+	then
+		echo "ERROR: ${_LIOTA_ETCDIR} already exists, but is not a directory.  This is a fatal error."
+		exit
+	fi
+
+	mkdir -p "${_LIOTA_ETCDIR}"
+
+	if [[ "$?" != "0" ]]
+	then
+		echo "ERROR: Couldn't create ${_LIOTA_ETCDIR}.  This is a fatal error."
+		exit
+	fi
+fi
+
+# Deal with directory ownership
+if [[ -d "${_LIOTA_ETCDIR}" ]]
+then
+	# Change the ownership and group on the configuration directory
+	chown "${_LIOTA_ROOTUSER}":"${_LIOTA_USER}" "${_LIOTA_ETCDIR}"
+
+	if [[ "$?" != "0" ]]
+	then
+		echo "ERROR: Couldn't set the ownership and group to ${_LIOTA_ROOTUSER}:${_LIOTA_USER}.  This is a fatal error."
+		exit
+	fi
+
+
+	# Change the permissions on the configuration directory
+	chmod "${_LIOTA_ETCPERMS}" "${_LIOTA_ETCDIR}"
+
+	if [[ "$?" != "0" ]]
+	then
+		echo "ERROR: Couldn't set permissions to ${_LIOTA_ETCPERMS} on ${_LIOTA_ETCDIR}.  This is a fatal error."
+		exit
+	fi
+fi
+
+# Deal with copying in the example config file into place
+if [[ -e "${_LIOTA_CONFTO}" ]]
+then
+	echo "ERROR: There is a configuration file already existing at ${_LIOTA_CONFTO}, not going to overwrite. This is a fatal error."
+	exit
+fi
+
+if [[ ! -e "${_LIOTA_CONFFROM}" ]]
+then
+	echo "ERROR: ${_LIOTA_CONFFROM} does not seem to exist, nothing to copy.  This is a fatal error."
+	exit
+fi
+
+cp "${_LIOTA_CONFFROM}" "${_LIOTA_CONFTO}"
+
+if [[ "$?" != "0" ]]
+then
+	echo "ERROR: couldn't copy ${_LIOTA_CONFFROM} to ${_LIOTA_CONFTO}.  This is a fatal error."
+	exit
+fi
+
+chown "${_LIOTA_ROOTUSER}:${_LIOTA_USER}" "${_LIOTA_CONFTO}"
```

Comment:
User should handle this process manually(chown) as we are not sure if we can create a new "liota" user on the IoT Gateway.